### PR TITLE
Add Question Answering

### PR DESCRIPTION
**Pull request by seriousmac**
https://github.com/seriousmac/awesome-qa/blob/master/README.md

Question Answering (QA) is a specialized area in the field of Natural Language Processing (NLP) and Information Retrieval (IR). The QA systems are concerned with providing relevant answers in response to questions proposed in natural language.  Natural language QA is recognized as a capability with great potential. Traditionally, QA has attracted many AI researchers, but most QA systems developed are toy systems or games confined to lab and a very restricted domain. More recently, Many innovations have been made with the development of artificial intelligence.

Pull requests I reviewed:
#1412 
#1174 

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of sub-lists, preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

**Comment by aschrijver** [comment]
Hi @seriousmac ,

I reviewed your list and some things are missing:

- The `awesome` and `awesome-list` topics
- Wrong location of of awesome badge (should be next to header)
  - Note: You could add a CC0 license section like the one in this repo at the bottom of the README
- There are no contribution guidelines
- About QA is missing from ToC

Something I cannot judge about, but will leave to @sindresorhus is that this is not formatted like a common awesome list, but instead is more of an overview of the field..

**Comment by seriousmac** [comment]
Thanks for your delicate review comments. I applied all the terms you mentioned! :)

**Comment by seriousmac** [comment]
Thanks for your linting results. I have solved problems for all the terms awesome-lint mentioned! :)

File: readme.md [hunk 1]
```diff
@@ -281,6 +281,7 @@
 - [Speech and Natural Language Processing](https://github.com/edobashira/speech-language-processing#readme)
 	- [Spanish](https://github.com/dav009/awesome-spanish-nlp#readme)
 	- [NLP with Ruby](https://github.com/arbox/nlp-with-ruby#readme)
+	- [Question Answering](https://github.com/seriousmac/awesome-qa#readme)
```
Comment (MFatihMAR) [inline readme.md]: Also please provide a good description for the list:

> - Entries have a description, unless the title is descriptive enough by itself. It rarely is though.

**Comment by seriousmac** [comment]
I have reviewed 2 PRs:
#1412
#1174

File: readme.md (hunk 1 above)
Comment (seriousmac) [reply readme.md]: All about asking in natural language and answering with a machine

**Comment by seriousmac** [comment]
@MFatihMAR 
Would you mind if I ask to merge it?

File: readme.md [hunk 2]
```diff
@@ -281,6 +281,7 @@
 - [Speech and Natural Language Processing](https://github.com/edobashira/speech-language-processing#readme)
 	- [Spanish](https://github.com/dav009/awesome-spanish-nlp#readme)
 	- [NLP with Ruby](https://github.com/arbox/nlp-with-ruby#readme)
+	- [Question Answering](https://github.com/seriousmac/awesome-qa#readme) - All about asking in natural language and answering with a machine.
```
Comment (sindresorhus) [inline readme.md]: > The science of asking in natural language and answering with a machine.

?

**Comment by sindresorhus** [comment]
The title should be `awesome-qa` => `Awesome Question Answering`

---

I don't think the nested items (2nd level) in the `Contents` section adds much value and should be removed.

---

`types of QA` => `Types of QA`

`subtypes of QA` => `Subtypes of QA`

`paper` => `Paper`

`Anlaysis` => `Analysis`

---

The items in the `Codes` section should have descriptions.

---

> Entries have a description, unless the title is descriptive enough by itself. It rarely is though.

---

> Has consistent formatting and proper spelling/grammar.
The description starts with an uppercase character and **ends with a period.**

---

`Question Answering` in `
A curated list of the Question Answering (QA) subject which is a computer science discipline within` should be linkified to https://en.wikipedia.org/wiki/Question_answering

---


✖   24:1   ToC item "Running Competitions in QA" does not match corresponding heading "Competitions in QA"
✖  105:3   Link to https://github.com/sebastianruder/NLP-progress/blob/master/question_answering.md is dead     
✖  179:5   Link to https://pdfs.semanticscholar.org/presentation/6285/0c8902f8e2a079b2fe356a73a21cea55c9d2.pdf is dead 

---

The general formatting could use more work. Use proper capitalizations. Fix typos. Write more descriptions.

**Comment by seriousmac** [comment]
> The title should be `awesome-qa` => `Awesome Question Answering`
> 
> I don't think the nested items (2nd level) in the `Contents` section adds much value and should be removed.
> 
> `types of QA` => `Types of QA`
> 
> `subtypes of QA` => `Subtypes of QA`
> 
> `paper` => `Paper`
> 
> `Anlaysis` => `Analysis`
> 
> The items in the `Codes` section should have descriptions.
> 
> > Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
> 
> > Has consistent formatting and proper spelling/grammar.
> > The description starts with an uppercase character and **ends with a period.**
> 
> `Question Answering` in ` A curated list of the Question Answering (QA) subject which is a computer science discipline within` should be linkified to https://en.wikipedia.org/wiki/Question_answering
> 
> ✖ 24:1 ToC item "Running Competitions in QA" does not match corresponding heading "Competitions in QA"
> ✖ 105:3 Link to https://github.com/sebastianruder/NLP-progress/blob/master/question_answering.md is dead
> ✖ 179:5 Link to https://pdfs.semanticscholar.org/presentation/6285/0c8902f8e2a079b2fe356a73a21cea55c9d2.pdf is dead
> 
> The general formatting could use more work. Use proper capitalizations. Fix typos. Write more descriptions.

@sindresorhus 
I seem to have missed a lot of instructions. Thanks for your point.
I have updated according to your point of view.
I would appreciate it if you could confirm it once more.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/awesome/pull/1385/files#r249315993

**Comment by sindresorhus** [comment]
> The science of asking and answering in natural language with a machine.

👍

**Comment by sindresorhus** [comment]
Linkify `Question Answering (QA)` in `
A curated list of the Question Answering (QA) subject which is a computer science discipline within` to https://en.wikipedia.org/wiki/Question_answering

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1100439353736622081

**Comment by seriousmac** [comment]
I've fixed! Thanks a lot 👍

**Pull request merged**